File: 2 semester/Java labs/Lab_5/src/main/java/com/ua/lviv/iot/spares/Electronics.java
package com.ua.lviv.iot.spares;

import com.ua.lviv.iot.spares.CategoryEnum;
import com.ua.lviv.iot.spares.Spares;

public class Electronics extends Spares {
    private int type;

    public Electronics(final Integer id, final int serialNumber, final int price, final CategoryEnum detailCategory, final int type) {
        super(id, serialNumber, price, detailCategory);
        this.type = type;
    }


    @Override
    public final String toString() {
        return "Serial number: " + getSerialNumber()
                + "  price: " + getPrice()
                + "$  Category: " + getCategory()
                + "  detail type: " + type
                + "cm";
    }

    public String getHeaders() {
        return super.getHeaders() + "type ";
    }

    public String toCSV() {
        return super.toCSV() + "," + type;
    }

    public final int getType() {
        return type;
    }

    public void setType(int type){this.type = type;}

}

File: 2 semester/Python labs/lab2/spares/Electronics.py
from spares.Spares import *
from enum_category.SparesCategory import *


class Electronics(Spares):
    category = SparesCategory.ELECTRONICS

    def __init__(self, serial_number, price, type):
        self.serial_number = serial_number
        self.price = price
        self.type = type

    def __str__(self):
        return "Serial Number: " + str(self.serial_number) + "  Price: " + str(self.price) + " Category: " + str(self.category.value) + " Type: " + str(self.type)
